docs/source/notes/designing_new_logic.rst

```diff
@@ -0,0 +1,402 @@
+definition includes various sections.
+
+So far, the `push-button` type will only be recognized by the TextWorld Engine but not the Inform7 compiler.
+To fix that, a mapping between TWLL's type and I7L's counterpart - a `kind` - must be establish.
+
+Let's add the relevant I7L code for the `push-button`:
+
+.. code-block:: bash
+
+    type push-button : thing {
+
+        inform7 {
+            type {
+                kind :: "push-button";
+            }
+        }
+    }
+
+The above example presents the type in both languages. Another relevant and simple example is a built-in entity inside
+the TextWorld framework, called `door`. Below is the details of the door entity:
+
+.. code-block:: bash
+
+    # door
+    type door : thing {
+
+        inform7 {
+            type {
+                kind       :: "door";
+                definition :: "door is openable and lockable. A door is usually locked.";
+            }
+        }
+    }
+
+The designer, here, prefers to add two features to the `door` type, as openable and lockable. Any feature which is
+supposed to be as general feature of this type can be mentioned in `definition`. Moreover, if there is an initial state
+that the designer prefers to define to start the game with, should be mentioned in the definition part. For instance,
+the designer of our example game likes to have all doors in the game locked by default, when the game starts. Then "A
+door is usually locked." is added to definition. Similarly for a cloth, we can say "cloth are cleanable. A cloth is
+usually clean.".
+
+Each game usually include a few rooms with various entities in them. The combination of the initial state of each
+entity of the game (like a door which is either locked or open) creates the initial state of the game in general. The
+game continues state-to-state when the current state of any entity of the game changes, e.g. the locked door changes
+to unlocked (but closed). This is how the Text-based games perform.
+
+
+C. Defining a predicate
+-------------------------
+
+The next section of a logic file is a "Predicate", which describes various state definition of the logic of the entity
+using the TWLL. For instance, the states of a door can be either open/close or locked/unlocked. Another feature that
+can be defined as predicate is the connectivity of this entity with other entity(ies) in the environment. For example,
+a door can link two rooms, described as `link(r, door, r')`, or an object like a glass can be put on a supporter like a
+table, described as `on(o, s)`, which `o` stands for object type and `s` stands for supporter type. Any other feature
+which defines the state of the entity (in some way it also describes the state of the game), and designer would like
+to add it, should be described for the framework in TWLL format in the predicate part. Following is an example of the
+possible states of a door, in TWLL:
+
+.. code-block:: bash
```

Comment:
I don't think we should be using `bash` for these.  For one, the highlighter will assume that the `'` in `room'` opens a quoted string.

Reply:
@tavianator what should we use?